### ISSUE TITLE
Present calendar/tasks to Lightning as a single calendar?

### ISSUE THREAD

**Issue by a1291762** [jobisoft/TbSync#324]
## Your environment

TbSync version: 1.7.0.2
EAS-4-TbSync: 0.12
Thunderbird version: 60.9.0
OS: Ubuntu 18.04 (has not rolled out Thunderbird 68 yet)

## Expected behavior

Lightning allows me to create both calendar events and tasks and they synchronize with the server.

## Actual behavior

Because TbSync/EAS exposes the Office 365 Calendar and Tasks as separate calendars, care must be taken to only create calendar events in the Calendar calendar, and tasks in the Tasks calendar.

This is made difficult by Lightning's insistence on having only 1 default calendar. If I set the default to the Calendar calendar, new events will be created correctly but new tasks will fail to synchronize. If I set the default to the Tasks calendar, new tasks will be created correctly but new events will fail to synchronize.

I realize that the limitation of types comes from Office 365 and the assumption that every calendar supports every type comes from Lightning... but it seems to me that TbSync (or probably EAS?) could expose the Calendar and Tasks calendars as a single calendar to Lightning, thus avoiding this issue. Since the Calendar calendar never has tasks and the Tasks calendar never has events, it should be "easy" to figure out which items go to which Office 365 calendar.

I apologise if this would be more approprate filed against EAS.
I understand if this is not something that TbSync/EAS can easily do.
I have also filed a bug for Lightning since if it had support for calendars that do not support every item type, and support for item-specific defaults, that would also make things better. See https://bugzilla.mozilla.org/show_bug.cgi?id=1593853

**Comment by jobisoft** [comment]
I fixed that in TB68, as I now define calendars as "task" or "events" and they will only be used by lightning for the dedicated purpose. Can you try?

**Comment by a1291762** [comment]
Ubuntu 18.04 still hasn't released Thunderbird 68 but I downloaded the build from https://www.thunderbird.net/en-US/thunderbird/all/ and ran it.

The good news is that I can no longer create tasks in the Calendar calendar or events in the Tasks calendar. The bad news is that there is still only 1 default calendar and that causes repercussions.

If the default calendar is Calendar, the "quickly create new task" box is unusable.

![tasks](https://user-images.githubusercontent.com/327162/68347901-1af6b080-0144-11ea-8e97-1b9d59ef7c0f.png)

If I double click the list (or right click -> new task), the default calendar is Local, which is also not useful.

If the default calendar is Tasks, the "quickly create new task" box is useful. However, new events have Local as the default calendar, which is not useful.

Since in this case I don't actually need the Local calendar, I was able to remove it and select Tasks as the default calendar. This seems to give me useful behaviour... The "quickly create new task" box works and new events select Calendar as the default calendar (since it's now the only available event-supporting calendar).

I feel like there is something not quite right here... I shouldn't have to remove the Local calendar to make this work, but I suspect that may be more of a Lightning problem than a TbSync/EAS problem? (unless the "present both calendars as 1" idea is used to work around Lightning's limitation of 1 default calendar).
